src/profile-logic/merge-compare.js

```diff
@@ -1098,3 +1103,65 @@ function mergeMarkers(
 
   return { markerTable: newMarkerTable, translationMaps };
 }
+
+/**
+ * Merge screenshot markers from different threads. And update the target threads string table while doing it.
+ */
+function mergeScreenshotMarkers(
+  threads: Thread[],
+  targetThread: Thread
+): {
+  markerTable: RawMarkerTable,
+  translationMaps: TranslationMapForMarkers[],
+} {
+  const targetMarkerTable = { ...targetThread.markers };
+  const translationMaps = [];
+
+  threads.forEach(thread => {
+    if (thread.name === 'Compositor') {
```

Comment:
Btw, it would be better if we change this `if` to something more generic instead of hardcoding `Compositor` thread name here. The thread of screenshot markers is an implementation detail inside the Firefox. So someone can decide to put these markers inside another thread in the future. This check won't work in that case. We can replace it with this instead:
https://github.com/firefox-devtools/profiler/blob/510a546b4f4903d5987c575ace0fb5a50adadaf2/src/profile-logic/tracks.js#L365